graphql.go

```diff
@@ -73,12 +77,106 @@ func (c *Client) logf(format string, args ...interface{}) {
 	c.Log(fmt.Sprintf(format, args...))
 }
 
+// RetryConfig defines possible fields that client can supply for their retry strategies
+type RetryConfig struct {
+	// Optional - Max number of times client should retry
+	MaxTries int `json:"maxTries"`
+	// Required - Time interval to wait before trying attempt sending a request again
+	Interval float64 `json:"interval"`
+	// Required - Defines a policy to be used for retry
+	Policy PolicyType `json:"policy"`
+	// Optional - The max interval of time to wait before retrying
+	MaxInterval float64 `json:"maxInterval"`
+	// Optional - A mapping of statuses that client should retry.
+	// If not specifed, we will use default retry behavior on certain statuses
+	RetryStatus map[int]bool `json:"statusToRetry"`
+}
+
+// PolicyType defines a type of different possible Policies to be applied towards retrying
+type PolicyType string
+
+const (
+	// ExponentialBackoff - the interval is doubled after every try until hitting MaxInterval or MaxTries
+	ExponentialBackoff PolicyType = "exponential_backoff"
+	// Linear - the interval stays the same every try until hitting MaxTries
+	Linear PolicyType = "linear"
+)
+
+// Wrapper method to send request while optionally applying retry policy
+func (c *Client) sendRequest(req *http.Request) (*http.Response, error) {
+	var (
+		resp *http.Response
+		err  error
+	)
+
+	retryConfig := c.retryConfig
+	// Client did not specify retry config
+	if retryConfig.Policy == "" {
+		resp, err = c.httpClient.Do(req)
+		if err != nil {
+			return resp, fmt.Errorf("Error getting response: %v", err)
+		}
+		return resp, nil
+	}
+
+	for tryCount := 0; tryCount < retryConfig.MaxTries; tryCount++ {
+		resp, err = c.httpClient.Do(req)
+		if err == nil && !retryConfig.shouldRetry(resp.StatusCode) {
+			return resp, nil
+		}
+		log.Println("Will retry after interval expires")
+
+		// Wait for interval
+		log.Printf("Waiting for interval(%f) to expire...", retryConfig.Interval)
+		timer := time.NewTimer(time.Duration(retryConfig.Interval) * time.Second)
+		<-timer.C
```

Comment:
Should check here to see if the context has been cancelled when waiting for the next retry. You can use `select` with `case <-ctx.Done()` for that

Reply:
Will do this as well!